src/Symfony/Bundle/FrameworkBundle/Kernel/MicroKernelTrait.php

```diff
@@ -124,10 +124,15 @@ public function getBuildDir(): string
         return parent::getBuildDir();
     }
 
-    public function getShareDir(): string
+    public function getShareDir(): ?string
     {
         if (isset($_SERVER['APP_SHARE_DIR'])) {
-            return $_SERVER['APP_SHARE_DIR'].'/'.$this->environment;
+            if (false === $dir = filter_var($_SERVER['APP_SHARE_DIR'], \FILTER_VALIDATE_BOOL, \FILTER_NULL_ON_FAILURE) ?? $_SERVER['APP_SHARE_DIR']) {
```

Comment:
I'm not sure supporting `off` and the likes is useful. Why not "just" support the empty string here?